xarray/plot/plot.py

```diff
@@ -449,6 +450,21 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
         if imshow_rgb:
             # Don't add a colorbar when showing an image with explicit colors
             add_colorbar = False
+            # Calculate vmin and vmax automatically for `robust=True`
+            if robust:
+                vmin_tmp, vmax_tmp = darray.quantile(
+                    q=[ROBUST_PERCENTILE, 1 - ROBUST_PERCENTILE])
+                vmin = vmin if vmin is not None else vmin_tmp
+                vmax = vmax if vmax is not None else vmax_tmp
```

Comment:
perhaps we should skip the quantile calculation when both vmin/vmax are set. Then again, if robust is True, the behavior is poorly defined and maybe an error would be better.